### PR TITLE
feat: Scoped variable Support

### LINKED ISSUE #3917: Feature: Variables support
### 🔖 Feature description

User can upload and download manifest for creating scoped variables.
Variables  are supposed to be used in different entities across Devtron in upcoming releases.
User can start creating and saving variables in the system from UI by going to the scoped variables section in global configuration,  which will be exposed only if the flag `ENABLE_SCOPED_VARIABLES` is set to true, in the dashboard-cm.
On how to create variables, user can download the template which explains how to create a manifest

### 🎤 Pitch

NA

### ✌️ Solution

NA

### 🔄️ Alternative

_No response_

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/devtron-labs/devtron/blob/main/CODE_OF_CONDUCT.md)

AB#5534

### PR DESCRIPTION
**Pull request by subhashish-devtron**
# Description

Fixes #3917 

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [x] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Review by subhashish-devtron — COMMENTED** [review]
```const (
	PRIVATE = "private"
	PUBLIC  = "public"
)```
Please create a type

**Review by subhashish-devtron — COMMENTED** [review]
```
GetScopedVariables(scope models.Scope, varNames []string, isSuperAdmin bool) (scopedVariableDataObj []*models.ScopedVariableData, err error)
```
```isSuperAdmin -> maskSensitiveData```

File: pkg/variables/ScopedVariableService.go [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+
+	// getting all variables from cache
+	allVariableDefinitions := impl.VariableCache.GetData()
+
+	// cache is loaded and no active variables exist. Returns empty
+	if allVariableDefinitions != nil && len(allVariableDefinitions) == 0 {
+		return nil, nil
+	}
+
+	// Need to get from repo for isSensitive even if cache is loaded since cache only contains metadata
```
Comment (subhashish-devtron) [inline pkg/variables/ScopedVariableService.go:314]: please correct this comment

File: pkg/variables/utils/transformer.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+			if value.Selectors != nil && value.Selectors.AttributeSelectors != nil {
+				attribute.AttributeParams = value.Selectors.AttributeSelectors
+			}
+			attributes = append(attributes, attribute)
+		}
+		variable := models.Variables{
+			Definition: models.Definition{
+				VarName:          spec.Name,
+				DataType:         "primitive",
```
Comment (subhashish-devtron) [inline pkg/variables/utils/transformer.go:28]: constant?

**Review by kripanshdevtron — APPROVED** [review]

**Pull request merged**